posthog/test/test_migration_0222.py

```diff
@@ -0,0 +1,69 @@
+from posthog.test.base import TestMigrations
+
+
+class TagsTestCase(TestMigrations):
+
+    migrate_from = "0221_add_activity_log_model"  # type: ignore
+    migrate_to = "0222_fix_deleted_primary_dashboards"  # type: ignore
+    assert_snapshots = True
+
+    def setUpBeforeMigration(self, apps):
+        Organization = apps.get_model("posthog", "Organization")
+        Dashboard = apps.get_model("posthog", "Dashboard")
+        Team = apps.get_model("posthog", "Team")
+
+        org = Organization.objects.create(name="o1")
+
+        # CASE 1:
+        # Team with an existing primary dashboard that's not deleted isn't affected
+        # Expect: t1.primary_dashboard = d1
+        team_1 = Team.objects.create(name="t1", organization=org)
+        dashboard_1 = Dashboard.objects.create(name="d1", team=team_1)
+        team_1.primary_dashboard = dashboard_1
+        team_1.save()
+
+        # CASE 2:
+        # Team with a deleted primary dashboard and no other dashboards is set to None
+        # Expect: t2.primary_dashboard = None
+        team_2 = Team.objects.create(name="t2", organization=org)
+        dashboard_2 = Dashboard.objects.create(name="d2", team=team_2, deleted=True)
+        team_2.primary_dashboard = dashboard_2
+        team_2.save()
+
+        # CASE 3:
+        # Team with a deleted primary dashboard and other dashboards is set to the first dashboard
+        # Expect: t3.primary_dashboard = d4
+        team_3 = Team.objects.create(name="t3", organization=org)
+        dashboard_3 = Dashboard.objects.create(name="d3", team=team_3, deleted=True)
+        team_3.primary_dashboard = dashboard_3
+        team_3.save()
+        Dashboard.objects.create(name="d4", team=team_3)
```

Comment:
Nit: I think a case where "Team with a deleted primary dashboard and first non-deleted other dashboards is set to the first dashboard" would also be worth checking here. 

Case 4 might already check for this and I see no reason this would fail, but it would make sure that the same delete check works for teams with primary dashboards that are deleted.